subsys/suit/storage/src/suit_storage_nrf9280.c

```diff
@@ -0,0 +1,1085 @@
+#define SUIT_STORAGE_RAD_OFFSET		 FIXED_PARTITION_OFFSET(cpurad_suit_storage)
+#define SUIT_STORAGE_RAD_SIZE		 FIXED_PARTITION_SIZE(cpurad_suit_storage)
+#define SUIT_STORAGE_APP_ADDRESS	 suit_plat_mem_nvm_ptr_get(SUIT_STORAGE_APP_OFFSET)
+#define SUIT_STORAGE_APP_OFFSET		 FIXED_PARTITION_OFFSET(cpuapp_suit_storage)
+#define SUIT_STORAGE_APP_SIZE		 FIXED_PARTITION_SIZE(cpuapp_suit_storage)
+
+BUILD_ASSERT((SUIT_STORAGE_OFFSET <= SUIT_STORAGE_NORDIC_OFFSET) &&
+		     (SUIT_STORAGE_NORDIC_OFFSET + SUIT_STORAGE_NORDIC_SIZE <=
+		      SUIT_STORAGE_OFFSET + SUIT_STORAGE_SIZE),
+	     "Secure storage must be defined within SUIT storage partition");
+
+BUILD_ASSERT((SUIT_STORAGE_OFFSET <= SUIT_STORAGE_RAD_OFFSET) &&
+		     (SUIT_STORAGE_RAD_OFFSET + SUIT_STORAGE_RAD_SIZE <=
+		      SUIT_STORAGE_OFFSET + SUIT_STORAGE_SIZE),
+	     "Radiocore storage must be defined within SUIT storage partition");
+
+BUILD_ASSERT((SUIT_STORAGE_OFFSET <= SUIT_STORAGE_APP_OFFSET) &&
+		     (SUIT_STORAGE_APP_OFFSET + SUIT_STORAGE_APP_SIZE <=
+		      SUIT_STORAGE_OFFSET + SUIT_STORAGE_SIZE),
+	     "Application storage must be defined within SUIT storage partition");
+
+typedef uint8_t suit_storage_digest_t[32];
+
+struct suit_storage_nvv {
+	/* Manifest Runtime Non Volatile Variables. */
+	uint8_t var[EB_SIZE(suit_storage_nvv_t)];
+	/* Manifest Runtime Non Volatile Variables digest. */
+	uint8_t digest[EB_SIZE(suit_storage_digest_t)];
+};
+
+struct suit_storage_mpi_rad {
+	/* Radio recovery manifest MPI. */
+	uint8_t recovery[EB_SIZE(suit_storage_mpi_t)];
+	/* Radio local A manifest MPI. */
+	uint8_t local_a[EB_SIZE(suit_storage_mpi_t)];
+	/* Radio local B manifest MPI. */
+	uint8_t local_b[EB_SIZE(suit_storage_mpi_t)];
+	/* Radio MPI digest. */
+	uint8_t digest[EB_SIZE(suit_storage_digest_t)];
+};
+
+struct suit_storage_mpi_app {
+	/* Application root manifest MPI. */
+	uint8_t root[EB_SIZE(suit_storage_mpi_t)];
+	/* Application recovery manifest MPI. */
+	uint8_t recovery[EB_SIZE(suit_storage_mpi_t)];
+	/* Application local A manifest MPI. */
+	uint8_t local_a[EB_SIZE(suit_storage_mpi_t)];
+	/* Application local B manifest MPI. */
+	uint8_t local_b[EB_SIZE(suit_storage_mpi_t)];
+	/* Application local spare manifest MPI. */
+	uint8_t local_spare[EB_SIZE(suit_storage_mpi_t)];
+	/* Application MPI digest. */
+	uint8_t digest[EB_SIZE(suit_storage_digest_t)];
+};
+
+/* SUIT storage structure, that can be used to parse the NVM contents. */
+struct suit_storage_nordic {
+	/* Area accessible by the SECURE core debugger. */
+	union {
+		struct suit_storage_area_nordic {
+			/* Radio MPI Backup and digest. */
+			struct suit_storage_mpi_rad rad_mpi_bak;
+			/* Application MPI Backup and digest. */
+			struct suit_storage_mpi_app app_mpi_bak;
+
+			/* Nordic OEM Exposed SUIT Config Backup and digest. */
+			uint8_t oem_config_backup_digest[96];
+
+			/* The last SUIT update report. */
+			uint8_t report_0[160];
+
+			/* Reserved for Future Use. */
+			uint8_t reserved[1344];
```

Comment:
How is that possible?
Don't you want to define the cellular area separately?

Reply:
The current implementation is base on this layout. Please let me know if there is some misunderstanding.
https://github.com/SylwesterKonczyk/doc-internal/blob/suit_nrf92xx/sw_arch/suit/nrf92xx_topology.rst#4-layout-of-suit-storage

Reply:
I was surprised by the fact that it increased by over 1kB. Now I see why - you moved the Nordic-top to "Nordic, modem" area to merge all unused bytes into a single reserved region :+1: